runtime/vm/createramclass.cpp

```diff
@@ -2855,6 +2883,20 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 				allocationRequests[RAM_STATICS_FRAGMENT].prefixSize = 0;
 				allocationRequests[RAM_STATICS_FRAGMENT].alignedSize = 0;
 			}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			else if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_QTYPE)) {
+				/* For shared fragments, set alignedSize and prefixSize to 0 to make internalAllocateRAMClass() ignore them. */
+				allocationRequests[RAM_SUPERCLASSES_FRAGMENT].address = (UDATA *) companionClass->superclasses;
+				allocationRequests[RAM_SUPERCLASSES_FRAGMENT].prefixSize = 0;
+				allocationRequests[RAM_SUPERCLASSES_FRAGMENT].alignedSize = 0;
+				allocationRequests[RAM_INSTANCE_DESCRIPTION_FRAGMENT].address = companionClass->instanceDescription;
+				allocationRequests[RAM_INSTANCE_DESCRIPTION_FRAGMENT].prefixSize = 0;
+				allocationRequests[RAM_INSTANCE_DESCRIPTION_FRAGMENT].alignedSize = 0;
+				allocationRequests[RAM_STATICS_FRAGMENT].address = companionClass->ramStatics;
+				allocationRequests[RAM_STATICS_FRAGMENT].prefixSize = 0;
+				allocationRequests[RAM_STATICS_FRAGMENT].alignedSize = 0;
```

Comment:
Are methods ever called on the Ltype? AFAIK the ltype can never be instanciated. If we dont need methods, then we dont need the constant pool, itables, invoke cache, etc.

Reply:
> Are methods ever called on the Ltype? AFAIK the ltype can never be instanciated.

For the methods, I don't see they are called on Ltype now from bytecodes generated from javac. But I see the following in the [JEP](https://openjdk.org/jeps/401):

`The aconst_init instruction may refer to either a primitive type or a reference type. This determines whether a primitive value or a value object is produced.`

Reply:
I believe the `or a reference type.` was writtin in the context of Value classes which do not have a primitive type variant. Essentially, there can only be one instance type.

Reply:
Ah, I see.

Reply:
> Are methods ever called on the Ltype? AFAIK the ltype can never be instanciated. If we dont need methods, then we dont need the constant pool, itables, invoke cache, etc.

Do you want to set these fields to NULL on the Ltype ? I see we have a lot of places that use them directly without NULL check, for example:

https://github.com/eclipse-openj9/openj9/blob/6a2aac87ec442d1d96863265761996c9764b19f3/runtime/codert_vm/jswalk.c#L1005-L1008

https://github.com/eclipse-openj9/openj9/blob/6a2aac87ec442d1d96863265761996c9764b19f3/runtime/rasdump/heapdump.cpp#L717

https://github.com/eclipse-openj9/openj9/blob/6a2aac87ec442d1d96863265761996c9764b19f3/runtime/vmchk/checkclasses.c#L198-L199

https://github.com/eclipse-openj9/openj9/blob/6a2aac87ec442d1d96863265761996c9764b19f3/runtime/jvmti/jvmtiExtensionMechanism.c#L2556-L2562